cocos/rendering/custom/layout-graph-utils.ts

```diff
@@ -277,102 +279,230 @@ function getDescriptorBlockData (map: Map<string, DescriptorBlockData>, index: D
     return newBlock;
 }
 
+function getDescriptorGroupBlockData (map: Map<string, DescriptorBlockData>, index: DescriptorGroupBlockIndex): DescriptorBlockData {
+    const key = JSON.stringify(index);
+    const block = map.get(key);
+    if (block) {
+        return block;
+    }
+    const newBlock = new DescriptorBlockData(
+        index.descriptorType,
+        index.visibility,
+        0,
+        index.accessType,
+        index.viewDimension,
+        index.sampleType,
+        index.format,
+    );
+    map.set(key, newBlock);
+    return newBlock;
+}
+
+function getViewDimension (type: Type): ViewDimension  {
+    switch (type) {
+    case Type.SAMPLER1D:
+    case Type.TEXTURE1D:
+    case Type.IMAGE1D:
+        return ViewDimension.TEX1D;
+    case Type.SAMPLER2D:
+    case Type.TEXTURE2D:
+    case Type.IMAGE2D:
+        return ViewDimension.TEX2D;
+    case Type.SAMPLER2D_ARRAY:
+    case Type.TEXTURE2D_ARRAY:
+    case Type.IMAGE2D_ARRAY:
+        return ViewDimension.TEX2D_ARRAY;
+    case Type.SAMPLER_CUBE:
+    case Type.TEXTURE_CUBE:
+    case Type.IMAGE_CUBE:
+        return ViewDimension.TEXCUBE;
+    case Type.SAMPLER3D:
+    case Type.TEXTURE3D:
+    case Type.IMAGE3D:
+        return ViewDimension.TEX3D;
+    default:
+        return ViewDimension.UNKNOWN;
+    }
+}
+
 // make DescriptorSetLayoutData from effect directly
 export function makeDescriptorSetLayoutData (
     lg: LayoutGraphData,
     rate: UpdateFrequency,
     set: number,
     descriptors: EffectAsset.IDescriptorInfo,
+    isWebGPU: boolean,
 ): DescriptorSetLayoutData {
     const map = new Map<string, DescriptorBlockData>();
     const uniformBlocks: Map<number, UniformBlock> = new Map<number, UniformBlock>();
+
     for (let i = 0; i < descriptors.blocks.length; i++) {
         const cb = descriptors.blocks[i];
-        const block = getDescriptorBlockData(map, {
-            updateFrequency: rate,
-            parameterType: ParameterType.TABLE,
-            descriptorType: DescriptorTypeOrder.UNIFORM_BUFFER,
-            visibility: cb.stageFlags,
-        });
+        const block = isWebGPU
+            ? getDescriptorGroupBlockData(map, {
+                updateFrequency: rate,
```

Comment:
Since only web platforms may support WebGPU now, adding HTML5 check will make the treeshaking work for other platforms to reduce code size.

const block = HTML5 && isWebGPU ? getDescriptorGroupBlockData ...

Reply:
Good idea. I have never thought of.